src/main/scala/org/tensorframes/ExperimentalOperations.scala

```diff
@@ -45,6 +47,23 @@ trait ExperimentalOperations {
     }
     df.select(cols: _*)
   }
+
+  def appendShape(df: DataFrame, col: Column, shape: Array[Int]): DataFrame = {
+
+    val meta = new MetadataBuilder
+    val colDtypes = df.select(col).schema.fields.head.dataType
+    val basicDatatype = {
+      ExtraOperations.extractBasicType(colDtypes).getOrElse(throw new Exception(s"'$colDtypes' was not supported"))
+    }
+    meta.putString(MetadataConstants.tensorStructType,
+      SupportedOperations.opsFor(basicDatatype).sqlType.toString
+      )
+    meta.putLongArray(MetadataConstants.shapeKey, shape.map(_.asInstanceOf[Long]))
+    df.withColumn(col.toString(), col.as("", meta.build()))
+  }
+
+  def appendShape(df: DataFrame, col:Column, shape: util.ArrayList[Int]): DataFrame =
+    appendShape(df, col, shape)
```

Comment:
@yupbank you want to convert `shape` to an array, otherwise you get into a loop

Reply:
ah...that's why the stack overflowed! thanks a lot! saved my day!